README.md

```diff
@@ -1,111 +1,108 @@
+$ chmod +x nmc
+```
 
-This additional config must be provided in a YAML format mapping the logical name of the interface to its MAC address:
+### Manual method
 
-```yaml
-host_config:
-  - hostname: node1.example.com
-    interfaces:
-      - logical_name: eth0
-        mac_address: 00:10:20:30:40:50
-      - logical_name: eth1
-        mac_address: 10:20:30:40:50:60        
-  - hostname: node2.example.com
-    interfaces:
-      - logical_name: eth0
-        mac_address: 00:11:22:33:44:55
+```shell
+$ git clone https://github.com/suse-edge/nm-configurator.git
+$ cd nm-configurator
+$ cargo build --release # optionally specify --target flag if cross compiling
 ```
 
-**NOTE:** Interface names during the installation of nodes might differ from the preconfigured logical ones.
-This is expected and `nm-configurator` will rely on the MAC addresses and use the actual names for the
-NetworkManager configurations instead e.g. settings for interface with a predefined logical name `eth0` but
-actually named `eth0.101` will automatically be adjusted and stored to `/etc/NetworkManager/eth0.101.nmconnection`.
+## How to run it?
 
-## How to install it?
+### Generate config
 
-### Standard method:
+NMC depends on having the desired network state for all known nodes beforehand.
 
-Each release is published with `nm-configurator` already built for `amd64` and `arm64` Linux systems:
+[NetworkManager](https://documentation.suse.com/sle-micro/5.5/html/SLE-Micro-all/cha-nm-configuration.html)
+is using connection profiles defined as files stored under `/etc/NetworkManager/system-connections`.
+In order to generate these config (*.nmconnection) files, NMC uses the
+[nmstate](https://github.com/nmstate/nmstate) library and requires a configuration directory as an input.
+This directory must contain the desired network state for all hosts in a <i>hostname</i>.yaml file format.
+Please refer to the official nmstate docs for [examples](https://nmstate.io/examples.html).
 
-For AMD64 / x86_64 based systems:
 ```shell
-$ curl -o nm-configurator -L https://github.com/suse-edge/nm-configurator/releases/latest/download/nm-configurator-amd64 
-$ chmod +x nm-configurator
+$ export RUST_LOG=info # set log level ("error" by default)
+$ ./nmc generate --config-dir config --output-dir _out
+[2023-11-08T21:18:12Z INFO  nmc::generate_conf] Generating config from "config/node1.yaml"...
+[2023-11-08T21:18:12Z INFO  nmc::generate_conf] Generating config from "config/node2.yaml"...
+[2023-11-08T21:18:12Z INFO  nmc::generate_conf] Generating config from "config/node3.yaml"...
+[2023-11-08T21:18:12Z INFO  nmc] Successfully generated and stored network config
 ```
 
-For ARM64 based systems:
+The output is the following:
+
 ```shell
-$ curl -o nm-configurator -L https://github.com/suse-edge/nm-configurator/releases/latest/download/nm-configurator-arm64 
-$ chmod +x nm-configurator
-```
+$ ls -R _out
+_out:
+host_config.yaml  node1  node2  node3
```

Comment:
Are we missing instructions on how to build the `host_config.yaml` here?

Reply:
It is automatically generated. We list the results of `nmc generate` here.